src/extension.ts

```diff
@@ -11,15 +11,24 @@ const figures = require('figures');
 const opn = require('opn');
 
 export function activate(context: ExtensionContext) {
-	const cwd = workspace.rootPath;
+	let cwd = workspace.rootPath;
 
 	const disposable = commands.registerCommand('yo', () => {
+
+		//if a file is opened in text editor, use its folder as the generator root folder rather than the workspace.rootpath
+		if (window.activeTextEditor && window.activeTextEditor.document && window.activeTextEditor.document.fileName) {
+			cwd = path.dirname(window.activeTextEditor.document.fileName);
+		}
+		else {
+			cwd = workspace.rootPath;
+		}
+
```

Comment:
do not fix multiple issues in one PR. Want to review them separately. I'm not sure yet on this temporary solution.